frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/virtual-hardware-tab/virtual-hardware-tab.tsx

```diff
@@ -0,0 +1,263 @@
+import { VMWizardStorageBundle } from '../storage-tab/types';
+import { vmWizardStorageModalEnhanced } from '../storage-tab/vm-wizard-storage-modal-enhanced';
+import { StorageUISource } from '../../../modals/disk-modal/storage-ui-source';
+import { VHW_TYPES } from './types';
+
+import './virtual-hardware-tab.scss';
+
+const getVirtualStoragesData = (
+  storages: VMWizardStorageWithWrappers[],
+  pvcs: FirehoseResult<K8sResourceKind[]>,
+): VMWizardStorageBundle[] => {
+  const pvcLookup = createLookup(pvcs, getName);
+
+  return storages
+    .filter((storage) => _.includes(VHW_TYPES, storage.diskWrapper.getType()))
+    .map((wizardStorageData) => {
+      const {
+        diskWrapper,
+        volumeWrapper,
+        dataVolumeWrapper,
+        persistentVolumeClaimWrapper,
+      } = wizardStorageData;
+
+      const pvc = pvcLookup[volumeWrapper.getPersistentVolumeClaimName()];
+
+      const combinedDisk = new CombinedDisk({
+        diskWrapper,
+        volumeWrapper,
+        dataVolumeWrapper,
+        persistentVolumeClaimWrapper:
+          persistentVolumeClaimWrapper || (pvc && PersistentVolumeClaimWrapper.initialize(pvc)),
+        pvcsLoading: !isLoaded(pvcs),
+      });
+
+      let source = combinedDisk.getSourceValue();
+      let name;
+
+      if (source === StorageUISource.CONTAINER.getValue()) {
+        name = volumeWrapper.getContainerImage();
+        if (isWinToolsImage(volumeWrapper.getContainerImage())) {
+          source = 'Windows Tools';
+        }
+      }
+
+      if (source === StorageUISource.IMPORT_DISK.getValue()) {
+        name = volumeWrapper.getPersistentVolumeClaimName();
+      }
+
+      if (source === StorageUISource.URL.getValue()) {
+        name = dataVolumeWrapper.getURL();
+      }
+
+      return {
+        wizardStorageData,
+        source,
+        isCD: true,
+        name,
+        storageClass: combinedDisk.getStorageClassName(),
+      };
+    });
+};
+
+const VirtualHardwareTabFirehose: React.FC<VirtualHardwareTabFirehoseProps> = ({
+  wizardReduxID,
+  isLocked,
+  setTabLocked,
+  removeStorage,
+  storages,
+  persistentVolumeClaims,
+}) => {
+  const virtualStorages = getVirtualStoragesData(storages, persistentVolumeClaims);
+  const showStorages = virtualStorages.length > 0;
+  const disableAddCD = virtualStorages.length > 1;
+  const availableCDName = getAvailableCDName(storages.map((storage) => storage.disk));
+  const withProgress = wrapWithProgress(setTabLocked);
+
+  const addButtonProps = {
+    id: 'attach-cdrom',
+    className: 'pf-m-link--align-left',
+    onClick: () =>
+      withProgress(
+        vmWizardStorageModalEnhanced({
+          initialType: DiskType.CDROM,
+          blocking: true,
+          wizardReduxID,
+          availableCDName,
+        }).result,
+      ),
+    isDisabled: isLocked || disableAddCD,
+  };
+
+  return (
+    <div className="kubevirt-create-vm-modal__virtual-hardware-tab-container">
```

Comment:
nitpic but should we call this `kv-create-vm-wizard` seems more in line with the console and it's no longer a modal.

Reply:
@suomiy was against this naming before AFAIR

Reply:
are you referring to the `kv-` prefix or the swapping of `modal` for `wizard`? I think both of these are reasonable updates but @suomiy please correct me :)

Reply:
I am not against the `kv-` prefix if we want go this way. We should do a PR which changes this for the whole codebase.

+1 for wizard